internal/provider/resource_azure_peering_connection.go

```diff
@@ -0,0 +1,400 @@
+		Peering: &networkmodels.HashicorpCloudNetwork20200907Peering{
+			ID: peeringID,
+			Hvn: &sharedmodels.HashicorpCloudLocationLink{
+				ID:       hvnLink.ID,
+				Location: loc,
+			},
+			Target: &networkmodels.HashicorpCloudNetwork20200907PeeringTarget{
+				AzureTarget: &networkmodels.HashicorpCloudNetwork20200907AzurePeeringTarget{
+					Region:            peerVnetRegion,
+					ResourceGroupName: peerResourceGroupName,
+					SubscriptionID:    peerSubscriptionID,
+					TenantID:          peerTenantID,
+					VnetName:          peerVnetID,
+				},
+			},
+		},
+	}
+	log.Printf("[INFO] Creating peering connection between HVN (%s) and peer (%s)", hvnLink.ID, peerVnetID)
+	peeringResponse, err := client.Network.CreatePeering(peerNetworkParams, nil)
+	if err != nil {
+		return diag.Errorf("unable to create peering connection between HVN (%s) and peer (%s): %v", hvnLink.ID, peerVnetID, err)
+	}
+
+	peering := peeringResponse.Payload.Peering
+
+	// Set the globally unique id of this peering in the state now since it has
+	// been created, and from this point forward should be deletable
+	link := newLink(peering.Hvn.Location, PeeringResourceType, peering.ID)
+	url, err := linkURL(link)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(url)
+
+	peering, err = clients.WaitForPeeringToBePendingAcceptance(ctx, client, peering.ID, hvnLink.ID, loc, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[INFO] peering connection (%s) is now in PENDING_ACCEPTANCE state", peering.ID)
+
+	if err := setAzurePeeringResourceData(d, peering); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceAzurePeeringConnectionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	client := meta.(*clients.Client)
+
+	link, err := buildLinkFromURL(d.Id(), PeeringResourceType, client.Config.OrganizationID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	peeringID := link.ID
+	loc := link.Location
+
+	hvnLink, err := buildLinkFromURL(d.Get("hvn_link").(string), HvnResourceType, loc.OrganizationID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[INFO] Reading peering connection (%s)", peeringID)
+	peering, err := clients.GetPeeringByID(ctx, client, peeringID, hvnLink.ID, loc)
+	if err != nil {
+		if clients.IsResponseCodeNotFound(err) {
+			log.Printf("[WARN] peering connection (%s) not found, removing from state", peeringID)
+			d.SetId("")
+			return nil
+		}
+
+		return diag.Errorf("unable to retrieve peering connection (%s): %v", peeringID, err)
+	}
+
+	// The peering connection failed to provision properly so we want to let the user know and
+	// remove it from state
+	if peering.State == networkmodels.HashicorpCloudNetwork20200907PeeringStateFAILED {
+		log.Printf("[WARN] peering connection (%s) failed to provision, removing from state", peering.ID)
```

Comment:
It it normal in terraform to remove resource from state if it is in FAILED state? I would expect it to error out and expect user to take action manually rather than surprising them with deletion.

Reply:
Good question. So far we've taken the approach across our resources to not track the resource's `state` as a property - so the user doesn't have much recourse to fix anything in a bad state except to recreate. We delete with no error here so that we don't disrupt the rest of the TF apply. The user _will_ see that the peering is not there on the next TF run, because TF will show in the plan it still needs to be created.